modules/consensus/accept_test.go

```diff
@@ -701,3 +702,82 @@ func TestTaxHardfork(t *testing.T) {
 		t.Fatal("siafund pool was not increased correctly")
 	}
 }
+
+// mockGateway implements modules.Gateway to mock the Broadcast method.
+type mockGateway struct {
+	modules.Gateway
+	numBroadcasts int
+	mu            sync.RWMutex
+}
+
+func (g *mockGateway) Broadcast(string, interface{}) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	g.numBroadcasts++
+	return
+}
+
+// TestAcceptBlockBroadcasts tests that AcceptBlock broadcasts valid blocks and
+// that managedAcceptBlock does not.
+func TestAcceptBlockBroadcasts(t *testing.T) {
+	cst, err := blankConsensusSetTester("TestAcceptBlockBroadcasts")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst.Close()
+	mg := &mockGateway{
+		Gateway: cst.cs.gateway,
+	}
+	cst.cs.gateway = mg
+
+	// Test that Broadcast is called for valid blocks.
+	b, _ := cst.miner.FindBlock()
+	err = cst.cs.AcceptBlock(b)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Sleep to wait for possible calls to Broadcast to complete. We cannot
+	// wait on a channel because we don't know how many times broadcast has
+	// been called.
+	time.Sleep(1)
+	mg.mu.RLock()
+	numBroadcasts := mg.numBroadcasts
+	mg.mu.RUnlock()
+	if numBroadcasts != 1 {
+		t.Errorf("expected AcceptBlock to broadcast a valid block 1 time, instead it broadcasted %d times", mg.numBroadcasts)
+	}
+
+	// Test that Broadcast is not called for invalid blocks.
+	mg.mu.Lock()
+	mg.numBroadcasts = 0
+	mg.mu.Unlock()
+	err = cst.cs.AcceptBlock(types.Block{})
+	if err == nil {
+		t.Fatal("expected AcceptBlock to error on an invalid block")
+	}
+	// Sleep one second to wait for a possible call to g.Broadcast.
+	time.Sleep(1)
+	mg.mu.RLock()
+	numBroadcasts = mg.numBroadcasts
+	mg.mu.RUnlock()
+	if numBroadcasts != 0 {
+		t.Error("AcceptBlock broadcasted an invalid block")
+	}
+
+	// Test that Broadcast is not called in managedAcceptBlock.
+	mg.mu.Lock()
+	mg.numBroadcasts = 0
+	mg.mu.Unlock()
+	b, _ = cst.miner.FindBlock()
```

Comment:
You were asking earlier what the testminer does. `FindBlock` is a function of the testminer, and is only called during testing.